ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py

```diff
@@ -63,6 +63,23 @@ def _array_idxes_n_dtype(draw, **kwargs):
     return x, idxes, dtype
 
 
+@st.composite
+def _broadcastable_trio(draw):
+    shape = draw(helpers.get_shape(min_num_dims=1, min_dim_size=1))
+    cond = draw(helpers.array_values(dtype="bool", shape=shape))
+    dtypes, xs = draw(
+        helpers.dtype_and_values(
+            available_dtypes=['float32'],

```

Comment:
Should be able to call a strategy like [this](https://github.com/unifyai/ivy/blob/435c4b11bbd260bd4fd0e0bc826d7b73a4bed842/ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py#L421).

Reply:
I thought about it too, but when using such helper functions I get the following error:
```
RuntimeError: expected scalar type c10::Half but found float
E               Falsifying example: test_torch_where(
E                   broadcastables=(array([False]),
E                    [array([-1.], dtype=float16), array([-1.], dtype=float32)],
E                    ['float16', 'float32']),
E                   with_out=False,
E                   as_variable=[False],
E                   num_positional_args=0,
E                   native_array=[False],
E                   on_device='cpu',
E                   fn_tree='ivy.functional.frontends.torch.where',
E                   frontend='torch',
E               )
```
The error is due to a mixed dtype which is not allowed by `torch.where`. Not sure if I can set up a strategy to always have the same dtype but tests multiple dtypes at the same time.
By setting `available_dtypes=['float32']` when I run tests on my local machine everything works fine

Reply:
@simonetgordon pinging you just in case you missed my comment's notification. I'll wait a response before moving forward

Reply:
@LucaLumetti has this issue not been fixed by the changes to `torch.where`?

Reply:
@simonetgordon The problem arises when calling `torch.where` with two tensors of different types, e.g. `float16` and `float32`. Using the strategy you suggested, the function is called with tensors of different types thus resulting in an error.

Reply:
Hi @LucaLumetti, the `torch.where` promotes the dtypes of the input tensors so in principle this shouldn't be an issue:
```
def where(
    condition: torch.Tensor,
    x1: Union[float, int, torch.Tensor],
    x2: Union[float, int, torch.Tensor],
    /,
    *,
    out: Optional[torch.Tensor] = None,
) -> torch.Tensor:
    x1, x2 = ivy.promote_types_of_inputs(x1, x2)
    return ivy.astype(torch.where(condition, x1, x2), x1.dtype, copy=False)
```

 There may be something going awry in the testing function when `input_dtypes` is specified with two different `x` and `y` tensor dtypes. Perhaps try set the final two `input_dtypes` as one dtype derived from something like `x1, x2 = ivy.promote_types_of_inputs(x1, x2)`.